components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/DeviceManagementServiceProviderImpl.java

```diff
@@ -596,7 +596,48 @@ public int getDeviceCount() throws DeviceManagementException {
             int deviceCount = this.deviceDAO.getDeviceCount();
             return deviceCount;
         } catch (DeviceManagementDAOException e) {
-            throw new DeviceManagementException("Error occurred while obtaining devices all devices", e);
+            log.error("Error occurred while counting devices", e);
```

Comment:
Noted and fixed in commit [1].
[1] https://github.com/wso2/carbon-device-mgt/commit/cdbdc86f42b350c24e9d059ff7c23e1f09242df4